workflow/rocoto/gfs_tasks.py

```diff
@@ -552,93 +552,163 @@ def _fcst_cycled(self):
+        deps = []
+        data = f'{atm_anl_path}/{self.cdump}.t@Hz.atmanl.nc'
+        dep_dict = {'type': 'data', 'data': data, 'age': 120}
+        deps.append(rocoto.add_dependency(dep_dict))
+        data = f'{atm_anl_path}/{self.cdump}.t@Hz.sfcanl.nc'
+        dep_dict = {'type': 'data', 'data': data, 'age': 120}
+        deps.append(rocoto.add_dependency(dep_dict))
         data = f'{atm_anl_path}/{self.cdump}.t@Hz.loganl.txt'
-        dep_dict = {'type': 'data', 'data': data}
+        dep_dict = {'type': 'data', 'data': data, 'age': 60}
+        deps.append(rocoto.add_dependency(dep_dict))
+        dependencies = rocoto.create_dependency(dep=deps, dep_condition='and')
+        resources = self.get_resource('upp')
+        task = create_wf_task('atmanlupp', resources, cdump=self.cdump, envar=postenvars, dependency=dependencies,
+                              cycledef=self.cdump, command='&JOBS_DIR;/upp.sh')
+
+        return task
+
+    def atmanlprod(self):
+        postenvars = self.envars.copy()
+        postenvar_dict = {'FHRLST': '-f001'}
+        for key, value in postenvar_dict.items():
+            postenvars.append(rocoto.create_envar(name=key, value=str(value)))
+
+        atm_master_path = self._template_to_rocoto_cycstring(self._base["COM_ATMOS_MASTER_TMPL"])
+        deps = []
+        data = f'{atm_master_path}/{self.cdump}.t@Hz.master.grb2anl'
+        dep_dict = {'type': 'data', 'data': data, 'age': 120}
         deps.append(rocoto.add_dependency(dep_dict))
         dependencies = rocoto.create_dependency(dep=deps)
-        resources = self.get_resource('post')
-        task = create_wf_task('postanl', resources, cdump=self.cdump, envar=postenvars, dependency=dependencies,
-                              cycledef=self.cdump)
+        resources = self.get_resource('atmos_products')
+        task = create_wf_task('atmanlprod', resources, cdump=self.cdump, envar=postenvars, dependency=dependencies,
+                              cycledef=self.cdump, command='&JOBS_DIR;/atmos_products.sh')
 
         return task
 
-    def ocnpost(self):
-        if self.app_config.mode in ['forecast-only']:  # TODO: fix ocnpost in cycled mode
-            return self._post_task('ocnpost')
+    @staticmethod
+    def _get_ufs_postproc_grps(cdump, config):
 
-    def _post_task(self, task_name):
-        if task_name not in ['post', 'ocnpost']:
-            raise KeyError(f'Invalid post-processing task: {task_name}')
+        fhmin = config['FHMIN']
+        fhmax = config['FHMAX']
+        fhout = config['FHOUT']
 
-        def _get_postgroups(cdump, config):
+        # Get a list of all forecast hours
+        fhrs = []
+        if cdump in ['gdas']:
+            fhrs = range(fhmin, fhmax + fhout, fhout)
+        elif cdump in ['gfs']:
+            fhmax = np.max(
+                [config['FHMAX_GFS_00'], config['FHMAX_GFS_06'], config['FHMAX_GFS_12'], config['FHMAX_GFS_18']])
```

Comment:
Will the four different forecast lengths even work with this construct? Seems like it would just be whichever is the longest. Though I'm not sure we can do anything about it here without creating a different cycledef for each cycle; the best we could do is exit the job immediately if it was past the cycle-specific max.

Reply:
It does not.  I am not sure when/why `FHMAX_GFS_$cyc` was added.  Is there a use case for different early cycle forecast lengths?
For now, I would suggest to let this be and we can eliminate `FHMAX_GFS_$cyc` with a single `FHMAX_GFS` in a later iteration.

Reply:
I wrote this before I saw Jessica's issue that coincidentally also opened yesterday. Fine letting it be for now, just wanted to note it. We may need to restore the capability for GEFS though, which runs for sub-seasonal lengths at 00z only.